airflow/providers/elasticsearch/log/es_task_handler.py

```diff
@@ -125,9 +125,9 @@ def _render_log_id(self, ti: TaskInstance, try_number: int) -> str:
             else:
                 log_id_template = self.log_id_template
 
-        dag = ti.task.dag
-        assert dag is not None  # For Mypy.
         try:
+            dag = ti.task.dag
+            assert dag is not None  # For Mypy.
             data_interval: Tuple[datetime, datetime] = dag.get_run_data_interval(dag_run)
         except AttributeError:  # ti.task is not always set.
```

Comment:
Judging from the comment on the `except` clause, this should be

```python
try:
    dag = ti.task.dag
except AttributeError:  # ti.task is not always set.
    ...
else:
    assert dag is not None  # For Mypy.
    data_interval: Tuple[datetime, datetime] = dag.get_run_data_interval(dag_run)
```

instead.

Reply:
Fair request -- updated